Provide a default tracking with Font Size config

valtism opened the pull request:
We can already configure line-height with font sizes, but this allows
us to also specify the default tracking for these sizes too.

This can be overwritten with tracking utilities.

I asked here but got no response: #1874

adamwathan commented:
I like this idea a lot! My only hesitation is that bumping this up to 3 unnamed positional parameters is getting into "not obvious" territory in terms of what things do. I think it's probably _fine_ but I wonder if it would be worth introducing an object syntax at this point so things are more clear, like:

```js
module.exports = {
  theme: {
    // ...
    fontSize: {
      sm: ['14px', { lineHeight: '20px', letterSpacing: '0.01em' }],
      // ...
    }
  }
}
```

What do you think, is it too verbose for the people who would want this feature or is it worth it for the extra clarity?

estevanmaito commented:
I would vote for clarity in 10/10 cases. Better for documenting, even better because somebody wouldn't even need to look into the docs to know what the code above is doing.

valtism commented:
Sounds good. The one issue I see is that we will have to support both the current 2-parameter syntax and this new object syntax, which might be confusing in the documentation.

valtism committed:
Allow default letterSpacing in fontSize config

Extend the configuration syntax to allow specifying letterSpacing
alongside lineHeight in the fontSize config.

The syntax looks like
```
module.exports = {
  theme: {
    // ...
    fontSize: {
      sm: ['14px', { lineHeight: '20px', letterSpacing: '0.01em' }],
      // ...
    }
  }
}
```

Also retains backwards compatibility with array syntax of 1.3

valtism commented:
Changed to the suggested syntax

Larsklopstra commented:
Just wanted to make a PR and came across this, can't wait for it to be released 🎉

ilyosjon09 commented on src/plugins/fontSize.js:
```diff
@@ -15,6 +19,11 @@ export default function() {
               : {
                   'line-height': lineHeight,
                 }),
+            ...(letterSpacing === undefined
```
d

ilyosjon09 reviewed (commented):
u to

valtism commented:
@adamwathan Sorry to bother you, but are there plans for this to be included in any upcoming release?

valtism commented:
@Larsklopstra Pinging you to let you know this has been released
